From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Juan Sanchez Fuentes <juanesf91@gmail.com>
Date: Sun, 16 Nov 2025 13:44:00 +0000
Subject: sunxi: Add NVME boot target support

NVMe is the LAST boot target so SD (MMC), USB and eMMC are tried first.

Signed-off-by: Juan Sanchez Fuentes <juanesf91@gmail.com>
Assisted-by: opencode:big-pickle
---
 include/configs/sunxi-common.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index 111111111111..222222222222 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -208,6 +208,12 @@
 #define BOOT_TARGET_DEVICES_DHCP(func)
 #endif
 
+#ifdef CONFIG_CMD_NVME
+#define BOOT_TARGET_DEVICES_NVME(func) func(NVME, nvme, 0)
+#else
+#define BOOT_TARGET_DEVICES_NVME(func)
+#endif
+
 /* FEL boot support, auto-execute boot.scr if a script address was provided */
 #define BOOTENV_DEV_FEL(devtypeu, devtypel, instance) \
 	"bootcmd_fel=" \
@@ -224,7 +230,8 @@
 	BOOT_TARGET_DEVICES_SCSI(func) \
 	BOOT_TARGET_DEVICES_USB(func) \
 	BOOT_TARGET_DEVICES_PXE(func) \
-	BOOT_TARGET_DEVICES_DHCP(func)
+	BOOT_TARGET_DEVICES_DHCP(func) \
+	BOOT_TARGET_DEVICES_NVME(func)
 
 #ifdef CONFIG_OLD_SUNXI_KERNEL_COMPAT
 #define BOOTCMD_SUNXI_COMPAT \
-- 
Armbian
